apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx

```diff
@@ -19,28 +19,36 @@ const pageComponents = [
 ];
 
 const TeacherApplication = props => {
-  const {accountEmail, userId, schoolId} = props;
+  const {
+    // [MEG] TODO: remove allowPartialSaving prop when experiment is complete (TeacherApps will always have this option)
+    // instead, pass in allowPartialSaving prop to FormController
+    allowPartialSaving,
+    savedFormData,
+    accountEmail,
+    userId,
+    schoolId
+  } = props;
 
   const getInitialData = () => {
+    const dataOnPageLoad =
+      allowPartialSaving && savedFormData && JSON.parse(savedFormData);
```

Comment:
What do you see as the risk here? allowPartialSaving gets set in `teacher_application/new.js` and we don't want to look for savedFormData if we don't have partial saving on. But I'm curious what you're thinking

Reply:
Just for clarify, allowPartialSaving as a TeacherApplication prop should be deleted once the experiment is done. I think we'll still want it in FormController, but tbd

Reply:
sorry to be vague. I was thinking that it could eliminate weird states that bug bashers might get into if the experiment is not consistently on or off. for example, what happens if I have saved form data, but the next time I go to the site the experiment is not on, and then I try to complete and submit the application?  always checking for and displaying saved form data would avoid this situation. I could be convinced that this is not a big deal... it just seemed simplest to have the fewest number of states the user can get into.

Reply:
> Just for clarity, allowPartialSaving as a TeacherApplication prop should be deleted once the experiment is done. I think we'll still want it in FormController, but tbd

yeah, good point, this is probably unlikely to have a truly bad outcome in a bug bash, so it's probably fine to leave it as you have it. your choice :-)

Reply:
Ah, I was envisioning that if we turn off the experiment, we wouldn't want to load an application someone had started. But I could see that being very confusing for the user ... you've convinced me